pkg/controller/association/dynamic_watches.go

```diff
@@ -13,20 +17,32 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// esWatchName returns the name of the watch setup on the referenced Elasticsearch resource.
-func esWatchName(associated types.NamespacedName) string {
-	return associated.Namespace + "-" + associated.Name + "-es-watch"
+const (
+	esWatchNameTemplate           = "%s-%s-es-watch-%s"
+	esUserWatchNameTemplate       = "%s-%s-es-user-watch-%s"
+	associatedCAWatchNameTemplate = "%s-%s-ca-watch-%s"
+)
+
+var (
+	esWatchNameRegexp           = regexp.MustCompile(fmt.Sprintf(esWatchNameTemplate, "(.*)", "(.*)", `(\d+)`))
+	esUserWatchNameRegexp       = regexp.MustCompile(fmt.Sprintf(esUserWatchNameTemplate, "(.*)", "(.*)", `(\d+)`))
+	associatedCAWatchNameRegexp = regexp.MustCompile(fmt.Sprintf(associatedCAWatchNameTemplate, "(.*)", "(.*)", `(\d+)`))
```

Comment:
Wow. Never expected to see those regexp here 😄 . Why?
If we need to filter type / namespace / name and things like, can't we build data structures to express those?

Reply:
We could if we are willing to rewrite watch container (`DynamicEnqueueRequest`) to have keys other than strings 🙂  Alternatively, we could encapsulate those regexes in a struct that would serialize to a string key - is this what you've meant? Even then we still need a regex (see https://github.com/elastic/cloud-on-k8s/pull/4036#discussion_r544147617).

Reply:
Synced offline and agreed to do the approach we have with watching Secrets for SecureSettings - each reconciliation set all resources to watch under a simpler key.